lib/rules/unsafe-to-chain-command.js

```diff
@@ -1,47 +1,142 @@
+  'scrollTo',
+  'select',
+  'selectFile',
+  'spread',
+  'submit',
+  'type',
+  'trigger',
+  'uncheck',
+  'within',
+]
+
+/**
+ * @type {import('eslint').Rule.RuleMetaData['schema']}
+ */
+const schema = {
+  title: NAME,
+  description: DESCRIPTION,
+  type: 'object',
+  properties: {
+    methods: {
+      type: 'array',
+      description:
+        'An additional list of methods to check for unsafe chaining.',
+      default: [],
+    },
+  },
+}
+
+/**
+ * @param {import('eslint').Rule.RuleContext} context
+ * @returns {Record<string, any>}
+ */
+const getDefaultOptions = (context) => {
+  return Object.entries(schema.properties).reduce((acc, [key, value]) => {
+    if (!(value.default in value)) return acc
+
+    return {
+      ...acc,
+      [key]: value.default,
+    }
+  }, context.options[0] || {})
+}
+
+/** @type {import('eslint').Rule.RuleModule} */
 module.exports = {
   meta: {
     docs: {
-      description: 'Actions should be in the end of chains, not in the middle',
+      description: DESCRIPTION,
       category: 'Possible Errors',
       recommended: true,
       url: 'https://docs.cypress.io/guides/core-concepts/retry-ability#Actions-should-be-at-the-end-of-chains-not-the-middle',
     },
-    schema: [],
+    schema: [schema],
     messages: {
-      unexpected: 'It is unsafe to chain further commands that rely on the subject after this command. It is best to split the chain, chaining again from `cy.` in a next command line.',
+      unexpected:
+        'It is unsafe to chain further commands that rely on the subject after this command. It is best to split the chain, chaining again from `cy.` in a next command line.',
     },
   },
   create (context) {
+    const { methods } = getDefaultOptions(context)
+
     return {
       CallExpression (node) {
-        if (isRootCypress(node) && isActionUnsafeToChain(node) && node.parent.type === 'MemberExpression') {
-          context.report({ node, messageId: 'unexpected' })
+        if (
+          isRootCypress(node) &&
+          isActionUnsafeToChain(node, methods) &&
+          node.parent.type === 'MemberExpression'
+        ) {
+          context.report({
+            node,
+            messageId: 'unexpected',
+          })
         }
       },
     }
   },
 }
 
-function isRootCypress (node) {
-  while (node.type === 'CallExpression') {
-    if (node.callee.type !== 'MemberExpression') return false
-
-    if (node.callee.object.type === 'Identifier' &&
-        node.callee.object.name === 'cy') {
-      return true
-    }
+/**
+ * @param {import('estree').Node} node
+ * @returns {boolean}
+ */
+const isRootCypress = (node) => {
+  if (
+    node.type !== 'CallExpression' ||
+    node.callee.type !== 'MemberExpression'
+  ) {
+    return false
+  }
 
-    node = node.callee.object
+  if (
+    node.callee.object.type === 'Identifier' &&
+    node.callee.object.name === 'cy'
+  ) {
+    return true
   }
 
-  return false
+  return isRootCypress(node.callee.object)
```

Comment:
Refactored this to use a recursive function instead of mutating the original `node`.